src/gateway/Gateway.sol

```diff
@@ -72,7 +81,17 @@ contract Gateway is Auth, IGateway {
             msg.sender == investmentManager || msg.sender == poolManager || msg.sender == messages[message.toUint8(0)],
             "Gateway/invalid-manager"
         );
-        aggregator.send(message);
+
+        if (MessagesLib.messageType(message) == MessagesLib.Call.EntryPoint) {
```

Comment:
I'm confused, what do we actually need `entrypoint` for?

Reply:
The standard flow is
-  _EOA -> ERC7540Vault -> Investment Manager -> Gateway -> ..._

With the UX improvements and CentrifugeRouter introduction the flow is:
- _Protocol -> CentrifugeRouter -> ERC7540Vault -> Investment Manager -> Gateway -> ..._

When the tx is triggered through the CentrifugeRouter, there are funds passed along the way to pay for the tx costs on all adapters + dst chain.

There are couple of scenarios we would like to consider:
 -  The value corresponding to amount that needs to be paid to all bridges + dst chain ( tx cost ) is a pessimistic value. That means that is a majority of the cases it will higher than the actual amount. The remaining of the funds will be sent to the gateway thus the gateway will contain funds. Given this, we don't want users following first flow, take advantage of the remaining gateway funds and pay for them.
 - We would like to pre-pay for an user ( following first flow ) so we will send ETH to the gateway directly. We should allow the tx cost payment for that user.
 
 This is something me and @ilinzweilin  discussed. We agreed that it's important to differentiate the entry point even though it's not clear right now how it will work and what are the cases exactly where we going to need this.
 
 Let me know if you have questions

Reply:
I think it needs polishing. This is a rough implementation. Frankly I spent significant time how to compromise and do minimal changes to implement this ( use existing functionality ) and not introducing any new methods and code changes.